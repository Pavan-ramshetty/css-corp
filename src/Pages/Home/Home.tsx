import Product from 'components/Product';
import React, { useEffect, useState } from 'react';
import { ProductResponse } from 'types/ProductResponse';

type Props = {
  products: ProductResponse[];
  loadProducts: () => void;
  loadCart: () => void;
  loading: any;
};

const Home = ({ products, loadProducts, loadCart, loading }: Props) => {
  console.log(loading);

  useEffect(() => {
    loadProducts();
    loadCart();
  }, [loadProducts, loadCart]);

  return (
    <>
      {(loading['LOAD_PRODUCTS'] || loading['LOAD_CART']) && (
        <div className="flex justify-center items-center h-screen w-full absolute z-10 bg-gray-600 opacity-60">
          <h1 className="text-white text-4xl">Loading...</h1>
        </div>
      )}
      {products.map((product) => {
        return <Product key={product.id} {...product} />;
      })}
    </>
  );
};

export default Home;
