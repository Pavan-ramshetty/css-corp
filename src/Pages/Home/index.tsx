import Product from 'components/Product';
import { CartContext } from 'context/CartContext';
import React, { useContext, useEffect } from 'react';

interface Props {}

const Home = (props: Props) => {
  const {
    loadData,
    handleCart,
    updateCartItem,
    deleteCartItem,
    cartState: { products, cart, loading },
  } = useContext(CartContext);

  useEffect(() => {
    loadData();
  }, [loadData]);

  return (
    <>
      {loading && (
        <div className="flex justify-center items-center h-screen w-full absolute z-10 bg-gray-600 opacity-60">
          <h1 className="text-white text-4xl">Loading...</h1>
        </div>
      )}
      {products.map((product) => {
        const cartItem = cart.find((x) => x.productId === product.id);
        return (
          <Product
            key={product.id}
            handleCart={handleCart}
            cartItem={cartItem}
            updateCartItem={updateCartItem}
            deleteCartItem={deleteCartItem}
            {...product}
          />
        );
      })}
    </>
  );
};

export default Home;
