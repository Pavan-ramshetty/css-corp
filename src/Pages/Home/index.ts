import { AppDispatch, RootStore } from 'configureStore';
import { connect } from 'react-redux';
import { loadCartRequest, loadProductsRequest } from 'reducers/LoadingReducer';
import Home from './Home';

const mapStoreToProps = (store: RootStore) => {
  return {
    products: store.products,
    loading: store.loading,
  };
};

const mapDispatchToProps = (dispatch: AppDispatch) => {
  return {
    loadProducts: () => dispatch(loadProductsRequest()),
    loadCart: () => dispatch(loadCartRequest()),
  };
};

export default connect(mapStoreToProps, mapDispatchToProps)(Home);
