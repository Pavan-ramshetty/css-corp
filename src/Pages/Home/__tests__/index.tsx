import React from 'react';
import { render as rtlRender, screen } from '@testing-library/react';
import Home from '../Home';
import { Provider } from 'react-redux';
import store from '../../../configureStore';

function render(ui: React.ReactElement, { ...renderOptions } = {}) {
  const Wrapper: React.FC = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

describe('Test Home Page', () => {
  test('should take snapshot of loading', () => {
    const { container } = render(
      <Home
        products={[]}
        loading={{
          LOAD_PRODUCTS: true,
          LOAD_CART: true,
        }}
        loadCart={jest.fn()}
        loadProducts={jest.fn()}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should take snapshot of products', () => {
    const { container } = render(
      <Home
        products={[
          {
            id: 20,
            title: 'DANVOUY Womens T Shirt Casual Cotton Short',
            price: 12.99,
            description:
              '95%Cotton,5%Spandex, Features: Casual, Short Sleeve, Letter Print,V-Neck,Fashion Tees, The fabric is soft and has some stretch., Occasion: Casual/Office/Beach/School/Home/Street. Season: Spring,Summer,Autumn,Winter.',
            category: "women's clothing",
            image: 'https://fakestoreapi.com/img/61pHAEJ4NML._AC_UX679_.jpg',
            rating: {
              rate: 3.6,
              count: 145,
            },
          },
        ]}
        loading={{}}
        loadCart={jest.fn()}
        loadProducts={jest.fn()}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render home page', () => {
    render(
      <Home
        products={[]}
        loading={{
          LOAD_PRODUCTS: true,
        }}
        loadCart={jest.fn()}
        loadProducts={jest.fn()}
      />,
    );
    const loading = screen.queryByText(/Loading.../i);

    expect(loading).toBeInTheDocument();
  });

  test('should display products', () => {
    render(
      <Home
        products={[
          {
            id: 20,
            title: 'DANVOUY Womens T Shirt Casual Cotton Short',
            price: 12.99,
            description:
              '95%Cotton,5%Spandex, Features: Casual, Short Sleeve, Letter Print,V-Neck,Fashion Tees, The fabric is soft and has some stretch., Occasion: Casual/Office/Beach/School/Home/Street. Season: Spring,Summer,Autumn,Winter.',
            category: "women's clothing",
            image: 'https://fakestoreapi.com/img/61pHAEJ4NML._AC_UX679_.jpg',
            rating: {
              rate: 3.6,
              count: 145,
            },
          },
        ]}
        loading={{}}
        loadCart={jest.fn()}
        loadProducts={jest.fn()}
      />,
    );
  });
});
