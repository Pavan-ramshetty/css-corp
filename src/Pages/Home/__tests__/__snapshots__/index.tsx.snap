// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home Page should take snapshot 1`] = `
<div
  class="flex justify-center items-center h-screen w-full absolute z-10 bg-gray-600 opacity-60"
>
  <h1
    class="text-white text-4xl"
  >
    Loading...
  </h1>
</div>
`;

exports[`Test Home Page should take snapshot of loading 1`] = `
<div
  class="flex justify-center items-center h-screen w-full absolute z-10 bg-gray-600 opacity-60"
>
  <h1
    class="text-white text-4xl"
  >
    Loading...
  </h1>
</div>
`;

exports[`Test Home Page should take snapshot of products 1`] = `
<div
  class="w-full relative flex items-center bg-white px-4 pt-14 pb-8 overflow-hidden shadow-2xl sm:px-6 sm:pt-8 md:p-6 lg:p-8"
>
  <div
    class="w-full grid grid-cols-1 gap-y-8 gap-x-6 items-start sm:grid-cols-12 lg:gap-x-8"
  >
    <div
      class="aspect-w-2 aspect-h-3 rounded-lg bg-gray-100 overflow-hidden sm:col-span-4 lg:col-span-5"
    >
      <img
        alt="DANVOUY Womens T Shirt Casual Cotton Short"
        class="object-center object-cover"
        src="https://fakestoreapi.com/img/61pHAEJ4NML._AC_UX679_.jpg"
      />
    </div>
    <div
      class="sm:col-span-8 lg:col-span-7"
    >
      <h2
        class="text-2xl font-extrabold text-gray-900 sm:pr-12"
      >
        DANVOUY Womens T Shirt Casual Cotton Short
      </h2>
      <section
        aria-labelledby="information-heading"
        class="mt-2"
      >
        <h3
          class="truncate"
          id="information-heading"
        >
          95%Cotton,5%Spandex, Features: Casual, Short Sleeve, Letter Print,V-Neck,Fashion Tees, The fabric is soft and has some stretch., Occasion: Casual/Office/Beach/School/Home/Street. Season: Spring,Summer,Autumn,Winter.
        </h3>
        <p
          class="text-2xl text-gray-900"
        >
          $12.99
        </p>
        <div
          class="mt-6"
        >
          <h4>
            Reviews
          </h4>
          <div
            class="flex items-center"
          >
            <div
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="h-5 w-5 flex-shrink-0 text-gray-900"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="h-5 w-5 flex-shrink-0 text-gray-900"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="h-5 w-5 flex-shrink-0 text-gray-900"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="h-5 w-5 flex-shrink-0 text-gray-900"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="h-5 w-5 flex-shrink-0 text-gray-900 text-gray-200"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                />
              </svg>
            </div>
            <p>
              3.6
               out of 5 stars
            </p>
            <a
              class="ml-3 text-sm font-medium text-indigo-600 hover:text-indigo-500"
              href="#"
            >
              145
               reviews
            </a>
          </div>
        </div>
      </section>
      <section
        aria-labelledby="options-heading"
        class="mt-10"
      >
        <h3
          id="options-heading"
        >
          women's clothing
        </h3>
        <button
          class="mt-6 w-full bg-indigo-600 border border-transparent rounded-md py-3 px-8 flex items-center justify-center text-base font-medium text-white  focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 hover:bg-indigo-700 disabled:bg-gray-500"
          type="button"
        >
          Add to bag
        </button>
      </section>
    </div>
  </div>
</div>
`;
